File: frontend/src/types/discord.ts
export interface DiscordIntegration {
  uuid: string;
  name: string;
  webhook_configured: boolean;
  enabled: boolean;
  all_events: boolean;
  event_types: string[];
  status_filter: string[];
  category_filter: string[];
  name_terms: string[];
  created_at: string;
  updated_at: string;
}
export interface DiscordIntegrationInput {
  name: string;
  webhook_url?: string;
  enabled?: boolean;
  all_events?: boolean;
  event_types?: string[];
  status_filter?: string[];
  category_filter?: string[];
  name_terms?: string[];
}
export interface DiscordDeliveryHistory { uuid: string; discord_id: string; event_type: string; status_code: number; error: string; created_at: string }

File: backend/internal/entities/discord.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

// DiscordIntegration is an incoming-webhook destination. The URL is always
// encrypted at rest and is deliberately absent from API response types.
type DiscordIntegration struct {
	UUID                uuid.UUID
	Name                string
	EncryptedWebhookURL string
	Enabled             bool
	AllEvents           bool
	EventTypes          []string
	StatusFilter        []string
	CategoryFilter      []string
	NameTerms           []string
	CreatedAt           time.Time
	UpdatedAt           time.Time
}

type DiscordDeliveryHistory struct {
	UUID       uuid.UUID
	DiscordID  uuid.UUID
	EventType  string
	StatusCode int
	Error      string
	CreatedAt  time.Time
}
